packages/addon/lua/gelly/api/mods/manager.lua

```diff
@@ -1,86 +1,103 @@
+			repository.upsertMetadataForModId(mod.mod.ID, { enabled = DEFAULT_MOD == mod.mod.ID and true or false })
+		end)
+end
 
-	local info = include(path .. "/info.lua")
+--- Enables/disables a mod by its ID.
+---@param modId string
+---@param enabled boolean
+---@return nil
+function gellyx.mods.setModEnabled(modId, enabled)
+	local currentMetadata = repository.fetchMetadataForModId(modId)
 
-	if not info then
-		return INVALID_PARSE
+	if not currentMetadata then
+		logging.error(("Mod %s does not exist."):format(modId))
+		return
 	end
 
-	if
-		info.Name == nil
-		or info.Author == nil
-		or info.Description == nil
-		or info.Type == nil
-	then
-		return INVALID_PARSE
-	end
+	repository.upsertMetadataForModId(modId, { enabled = enabled })
+	logging.info("Mod %s is now %s.", modId, enabled and "enabled" or "disabled")
+	logging.info("Updating mod restrictions.")
 
-	print("[gellyx mods] - Loaded mod " .. info.Name)
-	print("\tAuthor: " .. info.Author)
-	print("\tDescription: " .. info.Description)
-	print("\tType: " .. info.Type)
-	print("\tPath: " .. path)
-
-	return {
-		Valid = true,
-		Enabled = true,
-		Info = info,
-		Path = path,
-		InitPath = path .. "/init.lua",
-	}
+	gellyx.mods.restrictModEntities()
 end
 
---- Loads a mod into the manager.
----@param path string The path to the mod.
----@return boolean If the mod was loaded successfully.
-function gellyx.mods.loadMod(path)
-	local mod = parseMod(path)
-
-	if not mod.Valid then
-		ErrorNoHalt("[gellyx mods] - Failed to load mod at " .. path)
-		return false
+local function getGlobalModConflicts()
+	-- two global mods can't be enabled at the same time
+	local globalMods = array(loadedMods)
+		:filter(function(mod)
+			return mod.Type == gellyx.mods.ModType.Global
+		end)
+		:map(function(mod)
+			return { info = mod, metadata = repository.fetchMetadataForModId(mod.ID) }
+		end)
+		:filter(function(mod)
+			return mod.metadata.enabled
+		end)
+		:map(function(mod)
+			return mod.info.ID
+		end)
+		:toArray()
+
+	if #globalMods > 1 then
+		return globalMods
 	end
 
-	gellyx.mods.loadedMods[mod.Info.Name] = mod
-
-	return true
+	return nil
 end
 
---- Loads all mods from the filesystem.
-function gellyx.mods.loadMods()
-	local _, dirs = file.Find("gelly/mods/*", "LUA")
+--- Runs all enabled mods.
+---@return nil
+function gellyx.mods.runMods()
+	hook.Run("GellyModsShutdown")
+
+	local globalModConflicts = getGlobalModConflicts()
 
-	for _, mod in ipairs(dirs) do
-		gellyx.mods.loadMod("gelly/mods/" .. mod)
+	if globalModConflicts then
+		logging.error(("Global mods %s are conflicting."):format(table.concat(globalModConflicts, ", ")))
+		return
 	end
-end
 
---- Initializes all loaded mods.
-function gellyx.mods.initializeMods()
-	for _, mod in pairs(gellyx.mods.loadedMods) do
-		if mod.Enabled then
+	array(loadedMods)
+		:filter(function(mod)
+			local enabled = repository.fetchMetadataForModId(mod.ID).enabled
+			return enabled
+		end)
+		:forEach(function(mod)
 			include(mod.InitPath)
```

Comment:
Could we not wrap this in a pcall to ensure that one mod failing doesn't destroy the entire addon?